codebuild/bin/install_s2n_head.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+set -ex
+pushd "$(pwd)"
+
+usage() {
+    echo "install_s2n_head.sh build_dir"
+    exit 1
+}
+
+if [ "$#" -ne "1" ]; then
+    usage
+fi
+
+BUILD_DIR=$1
+source codebuild/bin/jobs.sh
+cd "$BUILD_DIR"
+
+mkdir s2n_head
+cd s2n_head
+
+# Clone the most recent s2n commit
+git clone https://github.com/aws/s2n-tls
+
+mkdir ../build
+cd ../build
+
+cmake ../s2n_head/s2n-tls -DCMAKE_PREFIX_PATH="$BASE_S2N_DIR"/test-deps/openssl-1.1.1/
```

Comment:
Is there a way to not assume the openssl version?

Reply:
I'm not sure there's a way to not assume the openssl version. Maybe I could make it not assume the libcrypto?

Reply:
If you source s2n_setup_env.sh, there is a [LIBCRYPTO_ROOT](https://github.com/aws/s2n-tls/blob/main/codebuild/bin/s2n_setup_env.sh#L120) env variable

Reply:
also if you tell cmake the build dir, it will create and build against it:
```
cmake ./s2n_head -Bbuild ...
cmake --build ./s2n_head
```